### PR TITLE
MSPSDS-868 update existing product pages

### PR DESCRIPTION
**Pull request by MatthewBurstein**
- gtin => barcode
- product_type => category
- new field product_type
- new field webpage

### CODE REVIEW

File: mspsds-web/app/models/investigation.rb [hunk 1]
```diff
@@ -77,7 +77,7 @@ def as_indexed_json(*)
           only: %i[company_name company_number]
         },
         products: {
-          only: %i[batch_number brand description gtin model name]
+          only: %i[batch_number brand description product_code model name]
```
Comment (broder) [inline mspsds-web/app/models/investigation.rb]: Are these still the correct fields to index in elasticsearch?

File: mspsds-web/app/views/products/_form.html.slim [hunk 2]
```diff
@@ -2,45 +2,72 @@
 - text_field_width = search ? "govuk-!-width-full" : "govuk-!-width-one-half"
 
 fieldset.govuk-fieldset
-  legend.govuk-fieldset__legend.govuk-fieldset__legend--l
-    | Identifying labels
+  .govuk-form-group class=[text_field_width, ("govuk-form-group--error" if product.errors[:category].any?)]
+    = form.label :category, "Product category",
+            class: "govuk-fieldset__legend govuk-fieldset__legend--m govuk-!-margin-bottom-2"
+    span.govuk-hint
+      |  The high-level category (for reporting purposes)
```
Comment (broder) [inline mspsds-web/app/views/products/_form.html.slim]: Is the double space between `|` and `The` intentional?

File: mspsds-web/app/views/products/_form.html.slim [hunk 3]
```diff
@@ -2,45 +2,72 @@
...
+
+fieldset.govuk-fieldset
+  .govuk-form-group class=[text_field_width, ("govuk-form-group--error" if product.errors[:product_type].any?)]
+    = form.label :product_type, "Product type",
+            class: "govuk-fieldset__legend govuk-fieldset__legend--m govuk-!-margin-bottom-2"
+    span#report-product-type-hint.govuk-hint
+      |  In plain English, what is it?
+      br
+      |  For example, ‘dishwasher, soft toy, chainsaw...’
+    = form.text_field :product_type, class: "govuk-input search-term", aria_describedby: "report-product-type-hint"
```
Comment (broder) [inline mspsds-web/app/views/products/_form.html.slim]: Do we still need `.search-term`?

File: mspsds-web/app/assets/javascripts/creation_flow.js [hunk 4]
```diff
@@ -1,5 +1,6 @@
 /* globals simpleAccessibleAutocomplete */
 $(document).ready(function () {
   simpleAccessibleAutocomplete('hazard-type-picker', { showAllValues: true });
-  simpleAccessibleAutocomplete('product-type-picker', { showAllValues: true });
+  simpleAccessibleAutocomplete('product-category-picker', { showAllValues: true });
+  simpleAccessibleAutocomplete('location-autocomplete', { showAllValues: true });
```
Comment (broder) [inline mspsds-web/app/assets/javascripts/creation_flow.js]: Is this already added by `location_picker.js`?

File: mspsds-web/app/views/products/_form.html.slim [hunk 5]
```diff
@@ -2,45 +2,72 @@
...
 
 fieldset.govuk-fieldset
-  legend.govuk-fieldset__legend.govuk-fieldset__legend--l
-    | Identifying labels
+  .govuk-form-group class=[text_field_width, ("govuk-form-group--error" if product.errors[:category].any?)]
+    = form.label :category, "Product category",
+            class: "govuk-fieldset__legend govuk-fieldset__legend--m govuk-!-margin-bottom-2"
+    span.govuk-hint
+      |  The high-level category (for reporting purposes)
+    = form.select :category, product_categories, { include_blank: true },
```
Comment (broder) [inline mspsds-web/app/views/products/_form.html.slim]: Does this need a `aria_describedby` attribute?

File: mspsds-web/app/views/products/_product_card.html.slim [hunk 6]
```diff
@@ -0,0 +1,15 @@
+.govuk-grid-row.mspsds-product-card.govuk-body
+  .govuk-grid-column-one-half
+    span.govuk-caption-m[class="govuk-!-font-size-16"]
+      = "Product"
```
Comment (broder) [inline mspsds-web/app/views/products/_product_card.html.slim]: Could this just be `| Product`?

File: mspsds-web/app/views/products/_suggested_container.html.slim [hunk 7]
```diff
@@ -1,7 +1,7 @@
 h3.govuk-heading-l
```
Comment (broder) [inline mspsds-web/app/views/products/_suggested_container.html.slim]: Are we keeping the suggested products bit?

File: mspsds-web/app/views/products/index.html.slim [hunk 8]
```diff
@@ -6,18 +6,6 @@
...
-              class: "mspsds-search") do |form|
-        .mspsds-search-box
-          = form.label "Keywords", class: "govuk-label--s"
-          = form.text_field :q, class: "govuk-input"
-          = form.submit "Search", name: nil, class: "mspsds-search-box--submit"
-  .govuk-grid-column-three-quarters
-    p
-      /TODO remove the link below
-      = link_to "Create new product", new_product_path, class: "govuk-!-font-size-19 govuk-!-font-weight-bold"
+  .govuk-grid-column-full.mspsds-case-list-tab
```
Comment (broder) [inline mspsds-web/app/views/products/index.html.slim]: Should this be `.mspsds-product-list-tab`? Or do we need to make the class more generic?

File: mspsds-web/app/views/products/show.pdf.slim [hunk 9]
```diff
@@ -12,34 +12,34 @@ table[style="width:100%; text-align:left"]
       = @product.id
```
Comment (broder) [inline mspsds-web/app/views/products/show.pdf.slim]: Can we actually access this PDF file from anywhere?

File: mspsds-web/app/views/products/tabs/_details.html.slim [hunk 10]
```diff
@@ -8,41 +8,41 @@ hr.govuk-section-break.govuk-section-break--l
...
+            | Webpage
+          td.govuk-table__cell
+            = @product.webpage
         tr.govuk-table__row
           th.govuk-table__header[scope="row"]
             | Country of origin
           td.govuk-table__cell
             = @product.country_of_origin_for_display
-        tr.govuk-table__row
+          tr.govuk-table__row
```
Comment (broder) [inline mspsds-web/app/views/products/tabs/_details.html.slim]: Is this indentation correct?

File: mspsds-web/db/migrate/20190110143423_update_products_schema.rb [hunk 11]
```diff
@@ -0,0 +1,38 @@
+class UpdateProductsSchema < ActiveRecord::Migration[5.2]
+  class Product < ApplicationRecord; end
+  def change
+    safety_assured do
+      change_table :products do |t|
+        reversible do |dir|
+          dir.up do
+            t.rename :gtin, :product_code
+
+            t.rename :product_type, :category
```
Comment (broder) [inline mspsds-web/db/migrate/20190110143423_update_products_schema.rb]: Do we need to rename the `product_type` field on investigations as well?

File: mspsds-web/lib/tasks/rapex_import.rake [hunk 12]
```diff
@@ -46,12 +46,13 @@ def create_records_from_notification(notification, date)
...
-  Product.where.not(gtin: "").where(gtin: barcode_from_notification(notification)).first_or_create(
-    gtin: barcode_from_notification(notification),
+  Product.where.not(product_code: "").where(product_code: barcode_from_notification(notification)).first_or_create(
+    product_code: barcode_from_notification(notification),
     name: name,
     description: field_from_notification(notification, "description"),
     model: field_from_notification(notification, "type_numberOfModel"),
-    product_type: field_from_notification(notification, "category"),
+    category: field_from_notification(notification, "category"),
+    webpage: field_from_notification(notification, "webpage"),
```
Comment (broder) [inline mspsds-web/lib/tasks/rapex_import.rake]: Is `webpage` actually a field on the RAPEX export?

File: mspsds-web/app/views/products/show.pdf.slim (hunk 9 above)
Comment (MatthewBurstein) [reply mspsds-web/app/views/products/show.pdf.slim]: I don't think so. Have removed :)

File: mspsds-web/app/views/products/index.html.slim (hunk 8 above)
Comment (MatthewBurstein) [reply mspsds-web/app/views/products/index.html.slim]: Have removed all together in fact.

File: mspsds-web/db/migrate/20190110143423_update_products_schema.rb (hunk 11 above)
Comment (MatthewBurstein) [reply mspsds-web/db/migrate/20190110143423_update_products_schema.rb]: Yes have done so.

File: mspsds-web/app/models/investigation.rb (hunk 1 above)
Comment (MatthewBurstein) [reply mspsds-web/app/models/investigation.rb]: Have amended these following discussion with Ed.

File: mspsds-web/db/migrate/20190110143423_update_products_schema.rb [hunk 13]
```diff
@@ -0,0 +1,45 @@
+class UpdateProductsSchema < ActiveRecord::Migration[5.2]
+  class Product < ApplicationRecord; end
+  def change
+    safety_assured do # rubocop:disable Metrics/BlockLength
```
[the remark this answers is not in the document]
Comment (MatthewBurstein) [reply mspsds-web/db/migrate/20190110143423_update_products_schema.rb]: Thanks, that's interesting. Should I change the migration accordingly in this PR or is this just for my information?

File: mspsds-web/db/migrate/20190110143423_update_products_schema.rb [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+
+            t.remove :brand
+            t.remove :model
+          end
+
+          rename_column :investigations, :product_type, :product_category
+        end
+
+
+        dir.down do
```
[the remark this answers is not in the document]
Comment (MatthewBurstein) [reply mspsds-web/db/migrate/20190110143423_update_products_schema.rb]: I have now haha. It works as it should after a couple of minor tweaks

File: mspsds-web/db/migrate/20190110143423_update_products_schema.rb (hunk 13 above)
[the remark this answers is not in the document]
Comment (broder) [reply mspsds-web/db/migrate/20190110143423_update_products_schema.rb]: Mostly for your information. (Although we generally should try and follow their examples!)

File: mspsds-web/app/views/investigations/products/new.html.slim [hunk 15]
```diff
@@ -14,5 +14,3 @@
     .govuk-grid-row
       .govuk-grid-column-one-half
```
Comment (broder) [inline mspsds-web/app/views/investigations/products/new.html.slim]: Should these now be `govuk-grid-column-two-thirds`?

Comment (MatthewBurstein) [reply mspsds-web/app/views/investigations/products/new.html.slim]: I don't think so, where did you see this?

Comment (broder) [reply mspsds-web/app/views/investigations/products/new.html.slim]: There are lots of instances of `.govuk-grid-column-two-thirds` and the only instances of `govuk-grid-column-one-half` are on this suggested stuff.

Comment (MatthewBurstein) [reply mspsds-web/app/views/investigations/products/new.html.slim]: I'm not sure about other places but `one-half` matches the designs for the new flow.

Comment (broder) [reply mspsds-web/app/views/investigations/products/new.html.slim]: Fair enough, happy to leave it then
